Guided Investigation - MDATP Webshell Alerts.ipynb

```diff
@@ -0,0 +1,841 @@
+{
```

Comment:
<p>Just a general point on the notebook. It might be helpful for the user who wants to understand the code to have more commentary on what your doing.</p><p>I usually put text that is required to understand the output (and maybe some general stuff about what is happening) in a Markdown cell.</p><p>For "why I'm doing X" or "what am I doing here" I use code comments.</p><p> </p><p>I know that you have a lot of HTML output explaining things but it might be helpful to move some of this into static markdown so that someone opening the notebook for the first time has an idea what is going on without having to run it.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Azure/Azure-Sentinel-Notebooks/pull/28/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
<p>As the notebook is so dependant on prior output to allow progression, and it basically has to be run in order to not break, I moved a lot of the markdown into displaying programmatically to discourage the user from skipping ahead and running cells that would result in errors or inconsistent outputs. This comes at the cost of the notebook being harder to understand by less experienced users. </p><p> </p><p>Rather than moving things into markdown now, I think I'd be keen to build a version 1.1 that has better progress tracking through the notebook to ensure users have pre-requisite cells and made appropriate pre-requisite selections.</p>